### PR TITLE
feat: make `Http` fallible and introduce `HttpWithResult`

### LINKED ISSUE #9391: Unhandled `Http` exception
If you use the `Http` effect and handle it using `Http.handle` and you disconnect from the internet you get `java.net.ConnectException` from Java that is not handled by the `Http.handle` function. I think this should be handled gracefully by wrapping the response in a `Result`

### PR DESCRIPTION
**Pull request by sockmaster27**
Fixes #9391

### CODE REVIEW

**Comment by sockmaster27** [comment]
@jaschdoc
Can you check if this fixes what you were doing before?
I can't find anywhere in the documentation that a `ConnectException` can be thrown, but I've tried including it anyway.

File: main/src/library/Http.flix [hunk 1]
```diff
@@ -23,13 +23,25 @@ eff Http {
...
     import java.net.http.HttpRequest
     import java.net.http.{HttpRequest$BodyPublishers => BodyPublishers}
     import java.net.URI
     import java.util.{List => JList}
 
+    ///
+    /// An error that can occur when performing HTTP operations.
+    ///
+    pub enum Error {
+        case Generic(String)
```
Comment (magnus-madsen) [inline main/src/library/Http.flix:42]: Instead of this, we should reuse our IoError that we already have.

**Comment by magnus-madsen** [comment]
It would be good if @jaschdoc could offer some more information.

**Comment by jaschdoc** [comment]
It works but the error message is just `null` so maybe we should add some more info manually? Maybe leave in the stack trace?

**Comment by magnus-madsen** [comment]
> It works but the error message is just `null` so maybe we should add some more info manually? Maybe leave in the stack trace?

No stack trace, but he should use ErrorKind and a proper error message.

**Comment by magnus-madsen** [comment]
Test failures

File: main/src/library/IoError.flix [hunk 2]
```diff
@@ -38,6 +38,7 @@ mod IoError {
         case NotDirectory,
         case UnexpectedEof,
         case Unsupported,
+        case ConnectionFailed,
```
Comment (magnus-madsen) [inline main/src/library/IoError.flix:41]: Keep sorted?

**Comment by magnus-madsen** [comment]
What is the status on this?

**Comment by magnus-madsen** [comment]
The compiler tests are still failing.

File: main/src/library/App.flix
Comment (sockmaster27) [inline main/src/library/App.flix:1]: What's the clean FP way of doing this in Flix?

Comment (magnus-madsen) [reply main/src/library/App.flix:1]: Lets ask @JonathanStarup

File: main/src/ca/uwaterloo/flix/api/Flix.scala [hunk 3]
```diff
@@ -260,6 +260,7 @@ class Flix {
     "Abort.flix" -> LocalResource.get("/src/library/Abort.flix"),
     "Clock.flix" -> LocalResource.get("/src/library/Clock.flix"),
     "Http.flix" -> LocalResource.get("/src/library/Http.flix"),
+    "HttpWithResult.flix" -> LocalResource.get("/src/library/HttpWithResult.flix"),
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/Flix.scala:263]: Can we do this in separate PR?

This one is supposed to fix a bug.

Comment (sockmaster27) [reply main/src/ca/uwaterloo/flix/api/Flix.scala:263]: Or, well, there's sort of a circular dependency, since this PR won't compile without this, and introducing `HttpWithResult` is weird when there's no error handling in it.

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/api/Flix.scala:263]: OK. Rename the PR then

**Review by magnus-madsen — APPROVED** [review]
